src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -497,22 +498,10 @@ public function processProductCustomization($product, $data)
         $shopList = Shop::getContextListShopID();
 
         /** Update the customization fields to be deleted in the next step if not used */
-        $updateQuery = 'UPDATE `' . _DB_PREFIX_ . 'customization_field` cf
-            SET cf.`is_deleted` = 1
-            WHERE
-            cf.`id_product` = ' . (int)$product->id . ' 
-            AND cf.`is_deleted` = 0 ';
-        if (!empty($customization_ids)) {
-            $updateQuery .= 'AND cf.`id_customization_field` NOT IN (' . implode(',', array_map('intval', $customization_ids)) . ')';
-        }
-        Db::getInstance()->execute($updateQuery);
-
-        $usedCustomizationIds = Db::getInstance()->executeS(
-            'SELECT cd.`index` FROM `'._DB_PREFIX_.'customized_data` cd 
-            LEFT JOIN `'._DB_PREFIX_.'customization_field` cf ON cf.`id_customization_field` = cd.`index`
-            WHERE cf.`id_product` = ' . (int)$product->id
-        );
-        array_walk($usedCustomizationIds, create_function('&$v', '$v = (int)$v["index"];'));
+        $product->updateCustomizationFieldsToDeleted($customization_ids);
+
+        $usedCustomizationIds = $product->getUsedCustomizationFieldsIds();
+        array_walk($usedCustomizationIds, array($this, 'customizationFieldsArrayAdapter'));
```

Comment:
You don't need this custom call. Just use :
```php
$usedCustomizationIds = array_column($usedCustomizationIds, 'index');
$usedCustomizationIds = array_map('intval', $usedCustomizationIds);
```
and then you can go for `array_merge` and `array_unique`